tests/player.c

```diff
@@ -536,6 +439,335 @@ START_TEST (test_play_audio_eos)
+{
+  GList *list;
+
+  for (list = gst_player_get_subtitle_streams (media_info);
+      list != NULL; list = list->next) {
+    GstPlayerStreamInfo *stream = (GstPlayerStreamInfo *) list->data;
+    GstPlayerSubtitleInfo *sub = (GstPlayerSubtitleInfo *) stream;
+
+    fail_unless_equals_string (
+        gst_player_stream_info_get_stream_type (stream), "subtitle");
+    fail_unless (gst_player_stream_info_get_tags (stream) != NULL);
+    fail_unless (gst_player_stream_info_get_caps (stream) != NULL);
+    fail_unless_equals_string (
+        gst_player_stream_info_get_codec (stream), "Timed Text");
+    fail_unless (gst_player_subtitle_info_get_language (sub) != NULL);
+  }
+}
+
+static void
+test_media_info_object (GstPlayer * player, GstPlayerMediaInfo * media_info)
+{
+  GList *list;
+
+  /* gloabl tag */
+  fail_unless (gst_player_media_info_is_seekable (media_info) == TRUE);
+  fail_unless (gst_player_media_info_get_tags (media_info) != NULL);
+  fail_unless_equals_string (
+      gst_player_media_info_get_title (media_info), "Sintel");
+  fail_unless_equals_string (
+      gst_player_media_info_get_container_format (media_info), "Matroska");
+  fail_unless (gst_player_media_info_get_image_sample (media_info) == NULL);
+  fail_unless( strstr (gst_player_media_info_get_uri (media_info), "sintel.mkv") != NULL);
+
+  /* number of streams */
+  list = gst_player_media_info_get_stream_list (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 10);
+
+  list = gst_player_get_video_streams (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 1);
+
+  list = gst_player_get_audio_streams (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 2);
+
+  list = gst_player_get_subtitle_streams (media_info);
+  fail_unless (list != NULL);
+  fail_unless_equals_int (g_list_length (list), 7);
+
+  /* test subtitle */
+  test_subtitle_info (media_info);
+
+  /* test audio */
+  test_audio_info (media_info);
+
+  /* test video */
+  test_video_info (media_info);
+}
+
+static void
+test_play_media_info_cb (GstPlayer * player, TestPlayerStateChange change,
+    TestPlayerState * old_state, TestPlayerState * new_state)
+{
+  if (change == STATE_CHANGE_MEDIA_INFO_UPDATED)
+    test_media_info_object (player, new_state->media_info);
+  else if (change == STATE_CHANGE_END_OF_STREAM)
+    g_main_loop_quit (new_state->loop);
+}
+
+START_TEST (test_play_media_info)
+{
+  GstPlayer *player;
+  TestPlayerState state;
+  gchar *uri;
+
+  memset (&state, 0, sizeof (state));
+  state.loop = g_main_loop_new (NULL, FALSE);
+  state.test_callback = test_play_media_info_cb;
+
+  player = test_player_new (&state);
+
+  fail_unless (player != NULL);
+
+  uri = gst_filename_to_uri (TEST_PATH "/sintel.mkv", NULL);
+  fail_unless (uri != NULL);
+  gst_player_set_uri (player, uri);
+  g_free (uri);
+
+  gst_player_play (player);
+  g_main_loop_run (state.loop);
```

Comment:
On errors and without getting any media-info, you would still succeed this test here